src/RequestHandlerSwooleRunner.php

```diff
@@ -362,10 +374,55 @@ private function getStaticResource(
         }
 
         $response->header('Content-Type', $this->cacheTypeFile[$staticFile]);
+
+        // Handle Gzip
+        if ($this->isGzipAvailable($request)) {
+            [$contentEncoding, $compressionEncoding] = $this->getCompressionEncoding($request);
+            if ($contentEncoding && $compressionEncoding) {
+                $data = gzcompress(file_get_contents($staticFile), $this->gzip, $compressionEncoding);
```

Comment:
This is potentially really heavy in terms of memory usage.

Does `Swoole\Http\Response` allow providing a stream resource to `sendfile()` and/or `end()`? If so, we could create a stream and apply a [compression filter](http://php.net/manual/en/filters.compression.php) instead, which would be more performant...

Reply:
`sendfile` and `end` doesn't support stream, I tried to compress image by stream compression filter and use `write` method return response, it's theoretically ok, but `write` method will automatically set an `Transfer-Encoding: chunked` header, and I need to response a `Transfer-Encoding: gzip, chunked` header, then body cannot decode in client.

Reply:
use `write()` method instead.